client/hosting/overview/controller.tsx

```diff
@@ -1,14 +1,28 @@
 import { Context as PageJSContext } from '@automattic/calypso-router';
+import i18n from 'i18n-calypso';
 import HostingOverview from 'calypso/hosting/overview/components/hosting-overview';
 import HostingActivate from 'calypso/hosting/server-settings/hosting-activate';
 import Hosting from 'calypso/hosting/server-settings/main';
+import { successNotice } from 'calypso/state/notices/actions';
 
 export function hostingOverview( context: PageJSContext, next: () => void ) {
 	context.primary = <HostingOverview />;
 	next();
 }
 
 export function hostingConfiguration( context: PageJSContext, next: () => void ) {
+	// Update the url and show the notice after a redirect
+	if ( context.query && context.query.hosting_features === 'activated' ) {
+		context.store.dispatch(
+			successNotice( i18n.translate( 'Hosting features activated successfully!' ), {
+				displayOnNextPage: true,
+			} )
+		);
+		// Remove query param without triggering a re-render
+		const newUrl = new URL( window.location.href );
+		newUrl.searchParams.delete( 'hosting_features' );
+		window.history.replaceState( null, '', newUrl.toString() );
```

Comment:
```suggestion
		window.history.replaceState( null, '', removeQueryArgs( window.location.href, 'hosting_features' ) );
```

I see we have this approach in other places, it looks a little bit clearer:

https://github.com/Automattic/wp-calypso/blob/7e593ecd88c0ce4b4a2bedcd59a95c9bdb53a83d/client/jetpack-cloud/sections/agency-dashboard/sites-overview/index.tsx#L299-L304